src/platform.ts

```diff
@@ -32,8 +32,10 @@ export class SharkIQPlatform implements DynamicPlatformPlugin {
       const serialNumbers = config.vacuums;
       if (!email || !password) {
         log.error('Login information must be present in config');
-      } else if (!serialNumbers) {
+        return;
+      } else if (!Array.isArray(serialNumbers) || serialNumbers.length === 0) {
```

Comment:
check this is an array. an advanced user may enter the config manually (rather than using the UI) and put a string rather than an array (which currently would be truthy and so the for loop later would run against a string and error.